src/main/java/dk/cngroup/lentils/service/ImageService.java

```diff
@@ -129,8 +129,10 @@ public Image getImageFromMultipartFile(final MultipartFile file) {
         return new Image(getFilePath(file));
     }
 
-    @Transactional
     public void deleteImage(final Image image) {
+        if (!Optional.ofNullable(image).isPresent()) {
```

Comment:
mám dotaz - proč se toto zabaluje do optionalu?  nestačilo by i (image == null) ? přijde mi to čitelnější

Reply:
no jo, máš pravdu - tady to asi není nutné. Já jsem tam chtěla nějak zamotat ty lambda výrazy, ale nepovedlo se mi to a pak už to  tak zůstalo. Teď jsem to zkusila trochu opravit, ale přijde mi to spíš složitější.:
Buď použiju 2 metody, v první otestuju jestli image není null, a volám druhou - to ale nevypadá moc ideálně, něco jako:

public void deleteImageWithCheck(final Image image) {
        Optional.ofNullable(image).ifPresent(this::deleteImageAndFile);
    }

    public void deleteImageAndFile(final Image image) {
        try {
            FileUtils.forceDelete(FileUtils.getFile(FileTreatingUtils.getUserDir() + image.getPath()));
        } catch (IOException e) {
            throw new FileNotDeletedException();
        }
        imageRepository.delete(image);
    }

Nebo tam nechám jenom tu druhou metodu, a budu ji volat z HintService. addImageToHintAndSave
Optional.ofNullable(actualImage).ifPresent(imageService::deleteImageAndFile);

To mi ale taky přijde špatně, ten test by se neměl provádět při volání, ale přímo v té metodě.

Nebo se na to vykašlu a nechám to, jak píše Pavel. Co myslíte, že je nejlepší?

Z toho .ifPresent() se asi nedají volat 2 různé metody, že? Že by se zvlášť mazal soubor, a zvlášť Image z DB.

Reply:
V tomhle případě bych oddělal Optional a nechal jen ten guard if s tím return jak máš ted.
Nebo nevím, vypadá to, že Optional tady nic nepřidává.

Reply:
OK

Reply:
Ale dobrý, že sis to chtěla vyzkoušet :) 👍